src/components/BSelect/BSelect.vue

```diff
@@ -0,0 +1,230 @@
+<template>
+  <div>
+    <div>
+      <label>{{ selectLabel }} <span v-if=isRequired aria-label='Required'>*</span></label>
+    </div>
+    <div 
+      :aria-expanded="opened"
+      :aria-owns="id"
+      :aria-label="selectLabel"
+      :aria-required="isRequired"
+      role="combobox"
+      tabindex="0"
+      v-model="selectLabel"
+      :id="'dropdown' + id"
+      @keyup="keyUpHandler(selectedOption)"
+    >
+      <ul
+        :class="ulClass"
+        role="listbox"
+        :id="id"
+      >
+        <li 
+          :class="dropdownClass" 
+          @click="openList"
+          >{{ selectedOption ? selectedOption.text : placeholder }}
+          <span class="dropdown-arrow"><i class="fa fa-caret-down" aria-hidden="true"></i></span>
+        </li>
+        <li
+          v-if=opened
+          v-for="option in options"
+          :class="liClass"
+          :key="'selectOption' + option.id"
+          :id="'option' + option.id"
+          :value=option.id
+          :aria-selected="selectedOption && selectedOption.id === option.id"
+          @click="selectOption(option)"
+          role="option"
+          :ref="'option' + option.id"
+          @keyup="keyUpHandler(option)"
+          tabindex='-1'
+        > {{ option.text }}
+        </li>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script>
+
+const options = []
+
+export default {
+  name: 'b-select',
+  props: {
+
+    id: {
+      type: String,
+      required: true
+    },
+    selectOptions: {
+      type: Array
+    },
+    isRequired: {
+      type: Boolean,
+      required: true
+    },
+    selectLabel: {
+      type: String,
+      required: true
+    },
+    initialValue: {
+      type: Object,
+      required: false
+    },
+    placeholder: {
+      type: String,
+      required: false,
+      default: 'Choose Option'
+    }
+  },
+
+  methods: {
+    openList () {
+      this.opened = !this.opened
+      if (this.opened) {
+        this.$nextTick( function () {
+          this.setFocus()
+        })
+      } else {
+        document.querySelector(`#dropdown${this.id}`).focus()
+      }
+    },
+    selectOption (option) {
+      this.selectedOption = option
+      this.opened = !this.opened
+      document.querySelector(`#dropdown${this.id}`).focus()
+    },
+    setFocus () {
+      if (this.selectedOption) {
+        const line = document.querySelector(`#${this.id} #option${this.selectedOption.id}`)
+        line.focus()
+      } else {
+        const line = document.querySelector(`#${this.id} #option${this.options[0].id}`)
+        line.focus()  
+      }
+    },
+    keyUpHandler (option) {
+      const target = event.target;
+      event.stopPropagation();
+      if(event.keyCode === 40) {
```

Comment:
Ok, I've been reading up a bunch to figure out the best way to clean up this function because it's long. What if we used the in-template bindings like this:

    <div @keyup.space="openHandler">
      <ul>
        <li @keyup.arrow-up="upHandler" @keyup.arrow-down="downHandler" @keyup.enter="selectHandler" @keyup.space="selectHandler"/>
      </ul>
    </div>

And then the specific functionality can be in isolated functions.

What do you think?

Reply:
I was going that way to start, but then switched it to the one key press handler that would handle the logic for all key presses... I can go either way, anyone else want to weigh in?